src/components/Loader/PeopleTable.tsx

```diff
@@ -0,0 +1,51 @@
+import { Person } from '../../types';
+import React, { useEffect, useState } from 'react';
+import { PersonLink } from './PersonLink';
+import { useParams } from 'react-router-dom';
+
+interface Props {
+  people: Person[];
+}
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const [isValidSlug, setIsValidSlug] = useState(true);
+
+  useEffect(() => {
+    if (!!slug && !people.find(p => p.slug === slug)) {
+      setIsValidSlug(false);
+    }
+
+    setTimeout(() => {
+      setIsValidSlug(true);
+    }, 2000);
```

Comment:
Using `setTimeout` here may lead to a potential issue. If the component unmounts before the timeout finishes, this will try to update the state of an unmounted component, which is a memory leak. You should clear the timeout in the cleanup function of the `useEffect` hook.